### ISSUE TITLE
Video support

### ISSUE THREAD

**Issue by bartbutenaers** [ibm-early-programs/node-red-contrib-media-utils#7]
Dear,

Currently all the nodes in this contribution are focussed on audio.  Are there any plans to add video/image functionality to this contribution?  Or will this suite stay dedicated to audio.

If video capabilities would be added some day, what would be the preferred way of working : extending the existing nodes (like ffmpeg-conversion) with video capabilities, or creating a new node that is focussed on video only?  Please let me know if I could be of any help in such developments (via a pull request).

Kind regards,
Bart Butenaers

**Comment by chughts** [comment]
If you need video as output, then please let us know what it is that you require. We will work out how it best fits in either into an existing node, or as a new node.

Of course it will be quicker if you write the updates and create a pull request.

**Comment by bartbutenaers** [comment]
Hi Soheel,

thanks for your fast response! Before I start coding, it would be nice if you could point me into the right direction.

- My first idea was to have two separate (converter) nodes, one for audio and one for video: to keep the node properties view simple for the user.  However if you e.g. load an mp4 file, you need to extract both audio and video at the same time, so I assume a single node for both audio and video would be better.  What is your point of view about this?
- Currently the list of formats is hardcoded (mp3 ...).  Would it be better to load all the supported formats automatically, so the user can pick whatever he likes?  Or do you think this would become uncluttered?
`Ffmpeg.getAvailableFormats ...`
- If I'm not mistaken, the converter node currently always uses a (real or temporary) file as input for ffmpeg: if the input payload contains a buffer, that will be stored into a temporary file.  In case of images, this would result in lot's of overhead, due to the large number of bytes (e.g. on my poor Raspberry Pi).  Is it ok for you to handle buffers by passing a readable stream (wrapping that buffer) to node-fluent-ffmpeg directly, instead of using temporary files?  Or do you think that this is not relevant (e.g. that the underlying nodejs node would create a temporary file from that stream anyway) ?

Thanks in advance !
Bart

**Comment by chughts** [comment]
My preference would be for an override on msg.ffmpeg.conversiontarget. Then the code that performs the conversion

```
            var stream  = fs.createWriteStream(info.path);
            ffmpeg(pathToFile)
              .format(config.format)
              .noVideo()
              .audioChannels(numChannels)
              .audioFrequency(frequency)
              .on('start', conversionStart)
              .on('error', conversionError)
              .on('end', conversionEnd)
              .output(stream)
              .run();
```

should allow 

`              .format(config.format)`

to be overridden.

**Comment by chughts** [comment]
Output format can be set by setting msg.format. I have taken out the .noVideo() call, and added in some video codec, pixel and stream settings. mp4 output has been tested. The existing audio outputs work as before.

Released in version 0.0.5

**Comment by bartbutenaers** [comment]
Hello Soheel,

Didn't want to be impolite, but I was rather busy during the holiday period.  My apologies that I didn't respond earlier.

Thanks a lot for adding the `msg.format` option !!!

Kind regards,
Bart
